packages/babel-plugin-jest-hoist/src/__tests__/hoistPlugin.test.ts

```diff
@@ -151,6 +151,32 @@ pluginTester({
       formatResult,
       snapshot: true,
     },
+    'jest.spyOn call on the imported module': {
+      code: formatResult(`
+        jest.mock('some-module', () => {
+          const module = jest.requireActual('some-module');
+          jest.spyOn(module, 'add');
+          return module;
+        });
+      `),
+      formatResult,
+      snapshot: true,
+    },
+    'jest.spyOn call in class constructor': {
+      code: formatResult(`
+        jest.mock('some-module', () => {
+          const Actual = jest.requireActual('some-module');
```

Comment:
```suggestion
          const module = jest.requireActual('some-module');
```

I think? `jest.spyOn(module, 'add');` seems wrong without it.

---

not that it matters, since what we're checking is the hoisting. but still

Reply:
Thanks for spotting this, I intended to write `jest.spyOn(this, 'add')` here. Such an expression must not be hoisted before `super()`, and that's also what our original code does, triggering the bug.

Reply:
right, makes sense 👍